cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperator.java

```diff
@@ -121,63 +121,32 @@ public KafkaConnectAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pf
         Promise<KafkaConnectStatus> createOrUpdatePromise = Promise.promise();
         String namespace = reconciliation.namespace();
 
-        Map<String, String> annotations = new HashMap<>(2);
-
         LOGGER.debugCr(reconciliation, "Updating Kafka Connect cluster");
 
         boolean connectHasZeroReplicas = connect.getReplicas() == 0;
 
-        KafkaClientAuthentication auth = kafkaConnect.getSpec().getAuthentication();
-        List<CertSecretSource> trustedCertificates = kafkaConnect.getSpec().getTls() == null ? Collections.emptyList() : kafkaConnect.getSpec().getTls().getTrustedCertificates();
-
         final AtomicReference<String> desiredLogging = new AtomicReference<>();
         connectServiceAccount(reconciliation, namespace, connect)
                 .compose(i -> connectInitClusterRoleBinding(reconciliation, namespace, kafkaConnect.getMetadata().getName(), connect))
-                .compose(i -> {
-                    if (isNetworkPolicyGeneration) {
-                        return networkPolicyOperator.reconcile(reconciliation, namespace, connect.getName(), connect.generateNetworkPolicy(isUseResources(kafkaConnect), operatorNamespace, operatorNamespaceLabels));
-                    } else {
-                        return Future.succeededFuture();
-                    }
-                })
-                .compose(i -> deploymentOperations.getAsync(namespace, connect.getName()))
-                .compose(deployment -> {
-                    if (deployment != null) {
-                        // Extract information from the current deployment. This is used to figure out if new build needs to be run or not.
-                        buildState.currentBuildRevision = Annotations.stringAnnotation(deployment.getSpec().getTemplate(), Annotations.STRIMZI_IO_CONNECT_BUILD_REVISION, null);
-                        buildState.currentImage = deployment.getSpec().getTemplate().getSpec().getContainers().get(0).getImage();
-                        buildState.forceRebuild = Annotations.hasAnnotation(deployment, Annotations.STRIMZI_IO_CONNECT_FORCE_REBUILD);
-                    }
-
-                    return Future.succeededFuture();
-                })
-                .compose(i -> connectBuild(reconciliation, namespace, build, buildState))
+                .compose(i -> isNetworkPolicyGeneration ?
+                        networkPolicyOperator.reconcile(reconciliation, namespace, connect.getName(), connect.generateNetworkPolicy(isUseResources(kafkaConnect), operatorNamespace, operatorNamespaceLabels)) :
+                        Future.succeededFuture()
+                )
```

Comment:
I think this way of writing makes it easier to scan down the code and see the different steps, since the `if` and `else` blocks catch your attention. It also aligns with the approach on line 184 although that can all fit on one line more easily. For the actual line of code personally I don't find much difference in readability but since I know ternary operators can be controversial I'm happy to put it back to the `if/else`. An alternative would be to put it in it's own function

Reply:
I would be fine with a separate method.